pallets/pallet-web3-names/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+	pub type BlockNumberFor<T> = <T as frame_system::Config>::BlockNumber;
+	pub type Web3NameOwnerOf<T> = <T as Config>::Web3NameOwner;
+	pub type Web3NameInput<T> = BoundedVec<u8, <T as Config>::MaxNameLength>;
+	pub type Web3NameOf<T> = <T as Config>::Web3Name;
+	pub type Web3OwnershipOf<T> =
+		Web3NameOwnership<Web3NameOwnerOf<T>, Deposit<AccountIdOf<T>, BalanceOf<T>>, BlockNumberFor<T>>;
+
+	pub(crate) type CurrencyOf<T> = <T as Config>::Currency;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	// Web3Name -> Ownership
+	#[pallet::storage]
+	#[pallet::getter(fn owner)]
+	pub type Owner<T> = StorageMap<_, Blake2_128Concat, Web3NameOf<T>, Web3OwnershipOf<T>>;
+
+	// Owner -> Web3Name
+	#[pallet::storage]
+	#[pallet::getter(fn names)]
+	pub type Names<T> = StorageMap<_, Blake2_128Concat, Web3NameOwnerOf<T>, Web3NameOf<T>>;
+
+	// Web3Name -> ()
+	#[pallet::storage]
+	#[pallet::getter(fn is_banned)]
+	pub type Banned<T> = StorageMap<_, Blake2_128Concat, Web3NameOf<T>, ()>;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The origin allowed to ban names.
+		type BanOrigin: EnsureOrigin<Self::Origin>;
+		/// The currency type to reserve and release deposits.
+		type Currency: ReservableCurrency<AccountIdOf<Self>>;
+		/// The amount of KILT to deposit to claim a name.
+		#[pallet::constant]
+		type Deposit: Get<BalanceOf<Self>>;
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// The min encoded length of a name.
+		#[pallet::constant]
+		type MinNameLength: Get<u32>;
+		/// The max encoded length of a name.
+		#[pallet::constant]
+		type MaxNameLength: Get<u32>;
+		/// The type of origin after a successful origin check.
+		type OriginSuccess: CallSources<AccountIdOf<Self>, Web3NameOwnerOf<Self>>;
+		/// The origin allowed to perform regular operations.
+		type RegularOrigin: EnsureOrigin<Success = Self::OriginSuccess, <Self as frame_system::Config>::Origin>;
+		/// The type of a web3 name.
+		type Web3Name: FullCodec + Debug + PartialEq + Clone + TypeInfo + TryFrom<Vec<u8>, Error = Error<Self>>;
+		/// The type of a web3 name owner.
+		type Web3NameOwner: Parameter + Default;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A new web3 name has been claimed.
+		Web3NameClaimed {
+			owner: Web3NameOwnerOf<T>,
+			name: Web3NameOf<T>,
+		},
+		/// A web3 name has been released.
+		Web3NameReleased {
+			owner: Web3NameOwnerOf<T>,
+			name: Web3NameOf<T>,
+		},
+		/// A web3 name has been banned.
+		Web3NameBanned { name: Web3NameOf<T> },
+		/// A web3 name has been unbanned.
+		Web3NameUnbanned { name: Web3NameOf<T> },
```

Comment:
events are always associated with a pallet. I think we don't need to prefix all of them with `Web3`